breadbox/breadbox/crud/types.py

```diff
@@ -503,9 +522,34 @@ def get_dimension_type_labels_by_id(
     return get_dimension_type_metadata_col(db, dimension_type_name, col_name="label")
 
 
+def get_given_id_to_dimension_id_mapping(
+    db: SessionWithUser, dimension_type_name: str
+) -> Dict[str, str]:
+    dimension_type = get_dimension_type(db=db, name=dimension_type_name)
+
+    if dimension_type is None:
+        raise ResourceNotFoundError(
+            f"Dimension type '{dimension_type_name}' not found. "
+        )
+    if dimension_type.dataset_id is None:
+        return {}
+
+    from ..models.dataset import Dimension
```

Comment:
I forgot that I haven't yet removed these non-top-imports. There's some circular dependencies that need to be sorted out. 

Now that all the tests are passing, I'll go back an reorganize the files to avoid the circular imports. (I'm thinking I'll create a new "search_index.py" under service to collect all functions related to the search index.

Reply:
Excellent, thank you! I'd been thinking of making a `search_index` service a while back, but had decided to hold off until things were refactored. Hopefully that will help prevent a lot of future circular imports as well 🤞